geopandas/testing.py

```diff
@@ -20,7 +20,7 @@ def _isna(this):
         return pd.isnull(this)
 
 
-def geom_equals(this, that):
+def geom_equals_mask(this, that):
```

Comment:
```suggestion
def _geom_equals_mask(this, that):
```

and need to adjust elsewhere as well, of course. 
The `geopandas.testing` module is public, so therefore let's maybe make those new helpers private to avoid people starting to rely on it.